File: app/src/main/java/app/we/go/emojidraw/arch/di/features/practice/PracticeComponent.java
package app.we.go.emojidraw.arch.di.features.practice;

import app.we.go.emojidraw.arch.di.scope.ActivityScope;
import app.we.go.emojidraw.features.practice.PracticeActivity;
import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@ActivityScope
@Subcomponent(modules = {PracticeMvpModule.class,
                PracticeConfigurationModule.class})
public interface PracticeComponent extends AndroidInjector<PracticeActivity> {

    @Subcomponent.Builder
    public abstract class Builder extends AndroidInjector.Builder<PracticeActivity> {}
}

File: app/src/main/java/app/we/go/emojidraw/arch/di/features/practice/PracticeComponent.kt
package app.we.go.emojidraw.arch.di.features.practice

import app.we.go.emojidraw.arch.di.ActivityScope
import app.we.go.emojidraw.features.practice.PracticeActivity
import dagger.Subcomponent

@ActivityScope
@Subcomponent(modules = [(PracticeMvpModule::class), (PracticeConfigurationModule::class)])
interface PracticeComponent {

    fun inject(practiceActivity: PracticeActivity)
}
